### PR TITLE
Add requirements.txt file

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

Add a requirements.txt file with the Python packages that are not in the Fedora repos.

#### Rationale:
Make it easier to install the python dependacies.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+xmldiff
+yamlpath
+json2html
+types-openpyxl
+types-PyYAML
+openpyxl-stubs
+cmakelint
+-r docs/requirements.txt
```
Comment (rhmdnd) [inline requirements.txt:8]: We'll also need:

```
$ ./utils/ansible_playbook_to_role.py                                                                                                                                                                                                                                                                                                                                                                                     Please install PyGithub, on Fedora it's in the python-PyGithub package.
```

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+xmldiff
+yamlpath
+json2html
+types-openpyxl
+types-PyYAML
```
Comment (rhmdnd) [inline requirements.txt:5]: Looks like we'll need:

- PyYAML
- jinja2
- pandas (although this is from the tool I'm writing, so I can add it there if this lands first)
- tables
- openpyxl
- pygments

File: requirements.txt (hunk 1 above)
Comment (Mab879) [reply requirements.txt:8]: That text should be changed to `python3-PyGithub` that package does exist in Fedora.

**Review by rhmdnd — COMMENTED** [review]
Thanks for proposing this, Matt.

File: requirements.txt (hunk 2 above)
Comment (rhmdnd) [reply requirements.txt:5]: Ok - good deal. Thanks.

**Comment by Mab879** [comment]
Helps address #7988

File: requirements.txt (hunk 2 above)
Comment (matejak) [reply requirements.txt:5]: Don't be shy and add them as well - if you use `pip install` on a system where part of the requirements are already satisfied, they are simply skipped.

**Comment by rhmdnd** [comment]
I have a patch for introducing a test-requirements.txt file, too https://github.com/ComplianceAsCode/content/pull/10487

Another approach we could take to help keep dependencies orthogonal would be to group scripts in directories. For example, we could have `utils/build/` and `utils/development/`. The scripts dedicated to build pipelines or function would live in `build/` and everything else necessary for helping contributors write and maintain content could live in `development/`.

From there, it would be relatively easy for contributors to have virtualenvs for dev and build separately (e.g., `utils/development/requirements.txt`). That may or may not be useful.

A side effect is that it would be nice to group like tools together, which may lead to efforts to consolidate functionality or consider moving to a unified, intuitive tool set for development.

**Comment by Mab879** [comment]
Succeed by #10487

**Pull request closed without merging**